### ISSUE TITLE
[Proposal] Remove `/` prefix in container names

### ISSUE THREAD

**Issue by cpuguy83** [docker/docker#29997]
All containers get this weird `/` prefix which we have to account for all over the code.
This was used, IIRC, for the graphdb but is not needed today (and arguable that it was ever really "needed").

When I removed graphdb I chose to keep it so as to minimize the change and potential pieces that could break at the time.

It would be really great to remove this.

**Comment by justincormack** [comment]
Yeah, it has cropped up in lots of questions and has had handling code added. Its time for it to go. If we wanted a name hierarchy, we could still have `a/b` without an initial `/`.

**Comment by thaJeztah** [comment]
Yup, would be glad to see it go as well; here's an earlier PR doing this; https://github.com/docker/docker/pull/11080

And related PR's / issues; https://github.com/docker/docker/pull/16577, https://github.com/docker/docker/pull/18524

**Comment by yongtang** [comment]
👍 #29007 is also related to the `/` naming issue.

**Comment by AkihiroSuda** [comment]
Is this still on plan?

Wondering it might break compatibility

**Comment by thaJeztah** [comment]
Yes, I think that's the reason attempts to remove this failed so far

**Comment by cpuguy83** [comment]
I haven't tried to remove it. I honestly don't think it should be a break... but I haven't looked at it in awhile.
